collect_app/src/main/java/org/odk/collect/android/javarosawrapper/FormController.java

```diff
@@ -345,6 +348,11 @@ private FormInstance getInstance() {
         return formEntryController.getModel().getForm().getInstance();
     }
 
+    public Stream<Entity> getEntities() {
+        Entities extra = formEntryController.getModel().getExtras().get(Entities.class);
+        return extra.getEntities().stream().map(entity -> new Entity(entity.dataset, entity.properties));
```

Comment:
Why do you convert it to Stream? Can't it be a list?

Reply:
Could be a `List`, but `Stream` is slightly more efficient as it means that the `map` won't execute until a "terminal" operation (like the `forEach` in `SaveFormToDisk`). It's a really minor saving (borderline insignificant), but feels weird to take it out now seeing as the API the caller interacts with is basically the same either way.